### ISSUE TITLE
HiddenSelect invalid markup (WCAG A 4.1.1 Parsing)

### ISSUE THREAD

**Issue by MashaSinkevich** [adobe/react-spectrum#4084]
# 🐛 Bug Report 

`HiddenSelect` generates one empty `option` which causes w3c validation error `"Error: Element option without attribute label must not be empty."`. To meet accessibility criteria "A" all HTML markup should be valid (see https://www.w3.org/WAI/WCAG21/quickref/#parsing).

## Steps to reproduce
1. copy markup from https://react-spectrum.adobe.com/react-aria/useSelect.html
2. past the following markup into https://validator.w3.org/ (validate by direct input):
```
<!DOCTYPE html>
<html lang="en"><head><title>test</title></head><body>

<div aria-hidden="true" style="border: 0px; clip: rect(0px, 0px, 0px, 0px); clip-path: inset(50%); height: 1px; margin: 0px -1px -1px 0px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"><input type="text" tabindex="0" style="font-size: 16px;"><label>Favorite Color<select size="10" tabindex="-1"><option></option><option value="$.0">Red</option><option value="$.1">Orange</option><option value="$.2">Yellow</option><option value="$.3">Green</option><option value="$.4">Blue</option><option value="$.5">Purple</option><option value="$.6">Black</option><option value="$.7">White</option><option value="$.8">Lime</option><option value="$.9">Fushsia</option></select></label></div>

```
</body></html>

## 🤔 Expected Behavior

No validation errors.

## 😯 Current Behavior
There is a validation error:
```
"Error: Element option without attribute label must not be empty.

[From line 5, column 344; to line 5, column 352](https://validator.w3.org/nu/#l5c352)

"><option></option>"
```

## 💁 Possible Solution

add the label "empty" to `HiddenSelect.tsx`, e.g. :
```
<select {...selectProps}>
   <option label="empty" />
   ...
</select>

```

**Comment by snowystinger** [comment]
I can see it in our docs page too. Thanks for reporting. 
![Screenshot 2023-02-17 at 10 16 35 AM](https://user-images.githubusercontent.com/698229/219746335-57bc7029-1e2d-4bbc-967e-83b9f96dc76a.png)

**Comment by reidbarber** [comment]
Closing for now, but we can re-open if this ever presents an accessibility or parsing issue in a real browser
